src/Login.jsx

```diff
@@ -0,0 +1,93 @@
+import { useState, useEffect, useContext } from 'react';
+import { Container, TextField, Button, Typography, Alert } from '@mui/material';
+import axios from 'axios';
+import { useNavigate, useLocation } from 'react-router-dom';
+import NavBar from './components/NavBar.jsx';
+import Footer from './components/Footer.jsx';
+import { FavoritesContext } from './contexts/FavoritesContext.jsx';
+import theme from './theme.jsx';
+import { ThemeProvider } from '@mui/material/styles';
+
+
+const Login = () => {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [error, setError] = useState('');
+    const navigate = useNavigate();
+    const [successMessage, setSuccessMessage] = useState('');
+    const location = useLocation();
+    const { setFavorites } = useContext(FavoritesContext);
+
+    useEffect(() => {
+        if (location.state && location.state.message) {
+            setSuccessMessage(location.state.message);
+        }
+    }, [location.state]);
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        try {
+            const response = await axios.post('/login', { username, password });
+            const userData = response.data;
+            const userId = userData.id;
+
+            // Store user data in session storage
+            sessionStorage.setItem('userData', JSON.stringify(userData));
```

Comment:
security wise it would be better to have some sort of session token here (either server generated or a jwt or some other fancy thing). The reason being is that users are able to modify session storage themselves, so they could essentially take advantage of this mechanism to pretend to be any other user.